yew/src/virtual_dom/mod.rs

```diff
@@ -343,14 +343,13 @@ impl Classes {
     /// Adds a class to a set.
     ///
     /// If the provided class has already been added, this method will ignore it.
-    pub fn push<T: Into<Self>>(&mut self, class: T) {
-        let classes_to_add: Self = class.into();
-        self.set.extend(classes_to_add.set);
+    pub fn push<T: Into<Cow<'static, str>>>(&mut self, class: T) {
+        self.set.insert(class.into());
```

Comment:
This, once again, violates the rule of splitting up multiple classes... We can't ensure that the given `class` value doesn't contain multiple classes.
I'm honestly not sure either what the best way is to go about this. If you want to keep it like this we need to explicitly state this in the documentation or even mark the function as unsafe (which is kind of silly)

Reply:
I can't find one good solution that fits all. I know you preferred the previous one but for me it's terrible to have to declare 2 times the &'static str in the code.

Reply:
(Well, the solution with the generics actually worked better because we don't mix String and &'static str)

Reply:
I don't understand what you mean... There was no `&'static` in the method signature before, your changes added that just now.
The only thing I'm kind of unhappy about with the previous implementation is that it converts it into `Classes` only for it to be dropped again. The compiler can surely optimize this away but it's still an eyesore.

Reply:
> (Well, the solution with the generics actually worked better because we don't mix String and &'static str)

We would've had to deal with a whole different can of worms if we continued on with generics.
Being able to mix string literals with owned strings is a good thing in my opinion. It's very rare that all of your classes are dynamically generated, very often it looks something like this: `("table", "table-cell", format!("size-{}", self.size))`. This wouldn't work at all with generics unless we force the first two classes into owned strings as well.

Reply:
You're right it does make sense to be able to do that in `html!`

But for my use case here outside `html!`: https://github.com/cecton/yewprint/blob/main/src/lib.rs#L74-L97 it's not good because I would have to have those `&'static str` in 2 different places.